src/cargo/core/compiler/custom_build.rs

```diff
@@ -231,14 +231,11 @@ fn build_work(cx: &mut Context<'_, '_>, unit: &Unit) -> CargoResult<Job> {
     for cfg in bcx.target_data.cfg(unit.kind) {
         match *cfg {
             Cfg::Name(ref n) => {
-                cfg_map.insert(n.clone(), None);
+                cfg_map.insert(n.clone(), Vec::new());
             }
             Cfg::KeyPair(ref k, ref v) => {
-                if let Some(ref mut values) =
-                    *cfg_map.entry(k.clone()).or_insert_with(|| Some(Vec::new()))
-                {
-                    values.push(v.clone())
-                }
+                let values = cfg_map.entry(k.clone()).or_insert_with(|| Vec::new());
```

Comment:
You can use `or_default()` here:

```suggestion
                let values = cfg_map.entry(k.clone()).or_default();
```